### DISCUSSION TITLE
Support for FTP

### DISCUSSION THREAD

**Discussion by majioa** [vcr/vcr#860] [Ideas]
Is any or will be the FTP protocol supported by VCR gem?

**Comment by krainboltgreene** [comment]
There was at one point! Really, the only issue is finding a way to mock the FTP interfaces.

**Comment by mcfiredrill** [comment]
None of the existing library hooks can handle it?

**Comment by krainboltgreene** [comment]
I haven't seen a direct reference to it, but since FTP is under Net it should be "that hard" to write a ftpmock library.

**Comment by synth** [comment]
Any updates on this?

**Comment by ch1ago** [comment]
looks like we're gonna have to be the ones to write it :)

@krainboltgreene where do you think'd be the starting point to write this?
Should we look inside VCR itself, or is it gonna be more on the webmock gem?

**Comment by krainboltgreene** [comment]
@thejamespinto One thing to know is that vcr is largely three pieces:

  1. Mock Adapter (this is where you come in)
  2. Formatting (the cassettes)
  3. Persistance Adapter (file, redis, etc)

I would start by talking to the webmock team and seeing if they've been approached by this before, otherwise I would look at VCR's internal adapter interface and see what's "expected" of your FTP mock.

**Comment by b-nik** [comment]
This feature would be great indeed!

For anyone coming here looking for a quick solution for simple cases, here's something I tried:
https://gist.github.com/b-nik/35ee176d7116f9610b8f7ad130507e7b
